sdks/java/io/debezium/src/main/java/org/apache/beam/io/debezium/KafkaSourceConsumerFn.java

```diff
@@ -360,6 +361,24 @@ static class OffsetHolder implements Serializable {
         @Nullable Integer fetchedRecords) {
       this(offset, history, fetchedRecords, null, -1L);
     }
+
+    @Override
+    public boolean equals(Object other) {
+      if (!(other instanceof OffsetHolder)) {
+        return false;
+      }
+      OffsetHolder otherOffset = (OffsetHolder) other;
+      return Objects.equals(offset, otherOffset.offset)
+          && Objects.equals(history, otherOffset.history)
```

Comment:
I see that history is a List<?> but in practice below it is a List<byte[]>. Is this not problematic? As it would compare byte array references instead of the content.

Reply:
Thanks, added logic to do deeper comparison and added a test of equality.